mlflow/metrics/utils/make_genai_metric.py

```diff
@@ -0,0 +1,206 @@
+            "or aligns with the reference or ground truth text that is considered "
+            "accurate and appropriate for the given input. The ground truth serves as "
+            "a benchmark against which the provided output is compared to determine the "
+            "level of accuracy and fidelity.",
+            grading_prompt="Correctness: If the answer correctly answer the question, below are the "
+            "details for different scores: "
+            "- Score 0: the answer is completely incorrect, doesn’t mention anything about the question "
+            "or is completely contrary to the correct answer. "
+            "- Score 1: the answer provides some relevance to the question and answer one aspect "
+            "of the question correctly. "
+            "- Score 2: the answer mostly answer the question but is missing or hallucinating on one "
+            "critical aspect. "
+            "- Score 4: the answer correctly answer the question and not missing any major aspect",
+            examples=[example],
+            model="gateway:/gpt4",
+            variables=["ground_truth"],
+            parameters={"temperature": 1.0},
+            greater_is_better=True,
+            aggregate_options=["mean", "variance", "p90"],
+        )
+    """
+    import pandas as pd
+    import pyspark
+
+    def eval_fn(eval_df: Union[pd.DataFrame, pyspark.sql.DataFrame]) -> MetricValue:
+        """
+        This is the function that is called when the metric is evaluated.
+        """
+
+        class_name = f"mlflow.metrics.utils.prompts.{version}.EvaluationModel"
+        try:
+            evaluation_model_class_module = _get_class_from_string(class_name)
+        except Exception as e:
+            if isinstance(e, ModuleNotFoundError):
+                raise MlflowException(
+                    f"Failed to find evaluation model for version {version}."
+                    f"Please check the correctness of the version",
+                    error_code=INVALID_PARAMETER_VALUE,
+                ) from None
+            else:
+                raise MlflowException(
+                    f"Failed to construct evaluation model {version}. Error: {e!r}",
+                    error_code=INTERNAL_ERROR,
+                ) from None
+
+        variables_dict = {}
+        for variable in variables:
+            if variable in eval_df.columns:
+                variable_value = eval_df[variable]
+                variables_dict[variable] = variable_value
+            else:
+                raise MlflowException(
+                    f"{variable} does not exist in the Eval DataFrame {eval_df.columns}."
+                )
+
+        variable_string = (
+            ""
+            if variables_dict is None
+            else "\n".join(
+                [
+                    f"Provided {variable}: {variable_value}"
+                    for variable, variable_value in variables_dict.items()
+                ]
```

Comment:
```suggestion
                f"Provided {variable}: {variable_value}"
                for variable, variable_value in variables_dict.items()
```

just a little optimization